apps/native/app/src/screens/document-detail/document-communications.tsx

```diff
@@ -0,0 +1,283 @@
+import { useCallback, useEffect, useMemo, useRef, useState } from 'react'
+import { useIntl } from 'react-intl'
+import { Animated, ListRenderItemInfo, RefreshControl } from 'react-native'
+import { NavigationFunctionComponent } from 'react-native-navigation'
+import { useSafeAreaInsets } from 'react-native-safe-area-context'
+import styled, { useTheme } from 'styled-components/native'
+import { useUser } from '../../contexts/user-provider'
+import {
+  DocumentComment,
+  useGetDocumentQuery,
+} from '../../graphql/types/schema'
+import { createNavigationOptionHooks } from '../../hooks/create-navigation-option-hooks'
+import { useConnectivityIndicator } from '../../hooks/use-connectivity-indicator'
+import { useDateTimeFormatter } from '../../hooks/use-date-time-formatter'
+import { useLocale } from '../../hooks/use-locale'
+import { useNavigationModal } from '../../hooks/use-navigation-modal'
+import {
+  Alert,
+  Button,
+  Container,
+  ListItemSkeleton,
+  TopLine,
+  Typography,
+} from '../../ui'
+import { ComponentRegistry } from '../../utils/component-registry'
+import { createSkeletonArr } from '../../utils/create-skeleton-arr'
+import {
+  DocumentListItem,
+  TOGGLE_ANIMATION_DURATION,
+} from './components/document-list-item'
+import {
+  FloatingBottomContent,
+  FloatingBottomFooter,
+} from './components/floating-bottom-footer'
+import { DocumentReplyScreenProps } from './document-reply'
+
+type FlatListItem = DocumentComment | { __typename: 'Skeleton'; id: string }
+
+const LAST_REPLY_HEIGHT = 100
```

Comment:
We do not know before hand the height of the last message body, so this is a arbitrary value